### ISSUE TITLE
Default EthereumTests in WalletCore produce an network invalid output.

### ISSUE THREAD

**Issue by Chevel** [trustwallet/wallet-core#2517]
**Describe the bug**
There are conflicting instructions on how the payload should be formed and as a result the JSON-RPC request for a EthereumSigningInput fails.

The code example from the TrustWallet core is:
```
    func testSignERC20Transfer() {
        let input = EthereumSigningInput.with {
            $0.chainID = Data(hexString: "01")!
            $0.nonce = Data(hexString: "00")!
            // txMode not set, Legacy is the default
            $0.gasPrice = Data(hexString: "09c7652400")! // 42000000000
            $0.gasLimit = Data(hexString: "0130B9")! // 78009
            $0.toAddress = "0x6b175474e89094c44da98b954eedeac495271d0f" // DAI
            $0.privateKey = Data(hexString: "0x608dcb1742bb3fb7aec002074e3420e4fab7d00cced79ccdac53ed5b27138151")!
            $0.transaction = EthereumTransaction.with {
                $0.erc20Transfer = EthereumTransaction.ERC20Transfer.with {
                    $0.to = "0x5322b34c88ed0691971bf52a7047448f0f4efc84"
                    $0.amount = Data(hexString: "1bc16d674ec80000")! // 2000000000000000000
                }
            }
        }
        let output: EthereumSigningOutput = AnySigner.sign(input: input, coin: .ethereum)

        XCTAssertEqual(output.encoded.hexString, "f8aa808509c7652400830130b9946b175474e89094c44da98b954eedeac495271d0f80b844a9059cbb0000000000000000000000005322b34c88ed0691971bf52a7047448f0f4efc840000000000000000000000000000000000000000000000001bc16d674ec8000025a0724c62ad4fbf47346b02de06e603e013f26f26b56fdc0be7ba3d6273401d98cea0032131cae15da7ddcda66963e8bef51ca0d9962bfef0547d3f02597a4a58c931")
        XCTAssertEqual(output.data.hexString, "a9059cbb0000000000000000000000005322b34c88ed0691971bf52a7047448f0f4efc840000000000000000000000000000000000000000000000001bc16d674ec80000")
    }
```

Whereas on the official website https://developer.trustwallet.com/wallet-core/integration-guide/wallet-core-usage#multi-coin-wallet
it is:
```
let input = EthereumSigningInput.with {
    $0.chainID = Data(hexString: "01")!
    $0.gasPrice = Data(hexString: "d693a400")! // decimal 3600000000
    $0.gasLimit = Data(hexString: "5208")! // decimal 21000
    $0.toAddress = "0xC37054b3b48C3317082E7ba872d7753D13da4986"
    $0.transaction = EthereumTransaction.with {
       $0.transfer = EthereumTransaction.Transfer.with {
           $0.amount = Data(hexString: "0348bca5a16000")!
       }
    }
    $0.privateKey = wallet.getKeyForCoin(coin: .ethereum).data
}
```

Nonce for example is missing in the web example, so we don't know whether it is optional.

The request using both examples on testnet (either as they are, or using ones own data) fails with the following error:
**invalid argument 0: json: cannot unmarshal hex string without 0x prefix into Go value of type hexutil.Bytes**

Based on searching for the error message the amount should not have leading zeros, which I have tried and does not work.
Some hexadecimals used in the examples (either on web or in code) use 0x prefix while others do not. This is inconsistent.

TLDR;
Inconsistent web docs and SDK examples.
Ambiguous expected hexadecimal representation.
No examples work when used in a testnet JSON-RPC call

**To Reproduce**
Attempt to perform a send raw transaction on the Goerli testnet.
https://ethereum.org/en/developers/docs/apis/json-rpc/#eth_sendrawtransaction

**Expected behavior**
I expected the JSON-RPC request to work using the tests template provide where my own data was replaced.
I expected the documentation to align with the code sample.

**Screenshots**
/

**Additional context**
The JSON-RPC calls for getBalance **DO** work. (since no EthereumSigningInput is used)
The JSON-RPC response is 200.
Based on that I am narrowing the problem down to the EthereumSigningInput parameters or output.

**Comment by Chevel** [comment]
Resolved by prepending "0x" to the output and using 0x for all parameters.

I would still love some cleaning up of the WalletCore and/or website.
